sdk/src/Core/Amazon.Runtime/Credentials/RefreshingAWSCredentials.cs

```diff
@@ -28,6 +28,20 @@ public abstract class RefreshingAWSCredentials : AWSCredentials, IDisposable
     {
         private Logger _logger = Logger.GetLogger(typeof(RefreshingAWSCredentials));
 
+        public override DateTimeOffset? Expiration
```

Comment:
A few classes inherit from `ImmutableCredentials` instead, and they have an `Expiration` / `Expires` property as well (e.g. [AssumeRoleImmutableCredentials](https://github.com/aws/aws-sdk-net/blob/main/sdk/src/Core/Amazon.Runtime/Credentials/AssumeRoleImmutableCredentials.cs) and [SSOImmutableCredentials](https://github.com/aws/aws-sdk-net/blob/main/sdk/src/Core/Amazon.Runtime/Credentials/_bcl45%2Bnetstandard/SSOImmutableCredentials.cs)).

Please correct my understanding if it's wrong, but I assumed I didn't have to change them since they're used to create new `CredentialsRefreshState` objects (which end up in the `currentState` field): https://github.com/aws/aws-sdk-net/blob/dabf1d1241924fa3bc8ebc5f55eaac99881b9c4d/sdk/src/Core/Amazon.Runtime/Credentials/AssumeRoleWithWebIdentityCredentials.cs#L211-L217

Reply:
But isn't the Expiration on these ImmutableCredentials sub class then redundant and possibly conflicting? I'm assuming these expiration properties were added on immutablecredentials because we didn't have the concept at AWSCredentials.

Reply:
I miss understood these were generated classes that we added a `partial` to make the be `AWSCredentials` objects.